timeless/poster/api.py

```diff
@@ -56,34 +57,42 @@ def send(self, **kwargs):
         return response
 
 
-class Authenticated(Poster):
+@attr.s
+class PosterAuthData:
+    """ Poster auth data class """
+    application_id = attr.ib()
+    application_secret = attr.ib()
+    redirect_uri = attr.ib()
+    code = attr.ib()
 
-    def __init__(self, **kwargs):
-        super().__init__(**kwargs)
 
-    def access_token(self):
-        """Fetches authorization tokens
-        :return: Token retrieved from Poster api or error returned by poster pi
+class Authenticated:
+    """ Poster Auth class """
+    def __init__(self, auth_data: PosterAuthData, **kwargs):
+        self.auth_url = "https://joinposter.com/api/v2/auth/access_token"
+        self.auth_data = auth_data
+
+    def auth(self):
         """
-        response = self.send(
-            method=self.POST,
-            action="auth/access_token"
-        )
+        Authentication into poster API
+        https://dev.joinposter.com/en/docs/api#authorization-in-api and use
+        """
+        auth_data = {
+            "application_id": self.auth_data.application_id,
+            "application_secret": self.auth_data.application_secret,
+            "grant_type": 'authorization_code',
+            "redirect_uri": self.auth_data.redirect_uri,
+            "code": self.auth_data.code,
+        }
+
+        response = requests.post(self.auth_url, data=auth_data)
+
         if not response.ok:
             raise Exception("Problem accessing poster api")
```

Comment:
@vladarefiev mb we will introduce a special expection type for errors related to Poster APIs? For example `class PosterAPIError(Exception)`

Reply:
@vryazanov Hmm, it's not my stuff, but i totally agree with you.